animation_nodes/nodes/spline/tilt_spline.py

```diff
@@ -10,22 +10,65 @@ class TiltSplineNode(bpy.types.Node, AnimationNode):
     bl_label = "Tilt Spline"
 
     useTiltList: VectorizedSocket.newProperty()
+    useSplineList: VectorizedSocket.newProperty()
+
     accumulateTilts: BoolProperty(name = "Accumulate Tilts", default = False,
         update = propertyChanged)
 
     def create(self):
-        self.newInput("Spline", "Spline", "spline", defaultDrawType = "PROPERTY_ONLY")
+        socket = self.newInput(VectorizedSocket("Spline", "useSplineList",
+            ("Spline", "spline"), ("Splines", "splines")))
+        socket.defaultDrawType = "PROPERTY_ONLY"
+        socket.dataIsModified = True
         self.newInput(VectorizedSocket("Float", "useTiltList",
             ("Tilt", "tilt"), ("Tilts", "tilts")))
-        self.newOutput("Spline", "Spline", "spline")
+
+        self.newOutput(VectorizedSocket("Spline", "useSplineList",
+            ("Spline", "spline"),
+            ("Splines", "splines")))
 
     def draw(self, layout):
         layout.prop(self, "accumulateTilts", text = "Accumulate")
 
-    def execute(self, spline, tilts):
-        if self.useTiltList:
-            _tilts = VirtualFloatList.fromList(FloatList.fromValues(tilts), 0)
+    def getExecutionFunctionName(self):
+        if self.useSplineList:
+            if self.useTiltList:
+                return "execute_MultipleSplines_MultipleTilts"
+            else:
+                return "execute_MultipleSplines_SingleTilt"
         else:
-            _tilts = VirtualFloatList.fromElement(tilts)
+            if self.useTiltList:
+                return "execute_SingleSpline_MultipleTilts"
+            else:
+                return "execute_SingleSpline_SingleTilt"
+
+    def execute_MultipleSplines_MultipleTilts(self, splines, tilts):
+        for spline in splines:
+            self.execute_SingleSpline_MultipleTilts(spline, tilts)
+        return splines
+
+    def execute_MultipleSplines_SingleTilt(self, splines, tilt):
+        for spline in splines:
+            self.execute_SingleSpline_SingleTilt(spline, tilt)
+        return splines
+
+    def execute_SingleSpline_SingleTilt(self, spline, tilt):
+        _tilts = VirtualFloatList.fromElement(tilt)
         tiltSplinePoints(spline, _tilts, self.accumulateTilts)
-        return spline
\ No newline at end of file
+        return spline
+
+    def execute_SingleSpline_MultipleTilts(self, spline, tilts):
+        _tilts = VirtualFloatList.fromList(self.prepareRadiusList(tilts, len(spline.points)), 0)
+        tiltSplinePoints(spline, _tilts, self.accumulateTilts)
+        return spline
+
+    def prepareRadiusList(self, tilts, length):
```

Comment:
Why is this function needed? Aren't we using a `VirtualFloatList`, which does what this function does?

Reply:
I also thought about this, but it is in the Set Spline Radius node.

Reply:
Used the `VirtualFloatList`.